components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/impl/SCIMUserManagerTest.java

```diff
@@ -550,4 +560,52 @@ public IObjectFactory getObjectFactory() {
 
         return new org.powermock.modules.testng.PowerMockObjectFactory();
     }
+
+    @Test
+    public void testGetEnterpriseUserSchema() throws Exception {
+
+        String tenantDomain = "carbon.super";
+        String externalClaimURI = "urn:ietf:params:scim:schemas:extension:enterprise:2.0:User";
+
+        Map<String, String> properties = new HashMap<String, String>() {{
+            put("ReadOnly", "true");
+            put("SupportedByDefault", "true");
+            put("Description", "sample");
+            put("Required", "true");
+        }};
+
+        Map<String, String> notSupportedByDefaultProperties = new HashMap<String, String>() {{
+            put("ReadOnly", "true");
+            put("SupportedByDefault", "false");
+            put("Description", "sample");
+            put("Required", "true");
+        }};
+
+        List<LocalClaim> localClaimMap = new ArrayList<LocalClaim>(){{
+            add(new LocalClaim("sample/department", new ArrayList<AttributeMapping>(), properties));
+            add(new LocalClaim("sample/organization", new ArrayList<AttributeMapping>(),
+                    notSupportedByDefaultProperties));
+            add(new LocalClaim("sample/manager", new ArrayList<AttributeMapping>(), properties));
+        }};
+
+        List<ExternalClaim> externalClaimMap = new ArrayList<ExternalClaim>(){{
+            add(new ExternalClaim(externalClaimURI, externalClaimURI+":department",
+                    "sample/department"));
+            add(new ExternalClaim(externalClaimURI, externalClaimURI+":organization",
+                    "sample/organization"));
+            add(new ExternalClaim(externalClaimURI, externalClaimURI+":manager",
+                    "sample/manager"));
+        }};
+
+        when(mockClaimMetadataManagementService.getExternalClaims(SCIMCommonConstants.
+                SCIM_ENTERPRISE_USER_CLAIM_DIALECT, tenantDomain)).thenReturn(externalClaimMap);
+        when(mockClaimMetadataManagementService.getLocalClaims(tenantDomain)).thenReturn(localClaimMap);
+        mockStatic(SCIMCommonUtils.class);
+        when(SCIMCommonUtils.isEnterpriseUserExtensionEnabled()).thenReturn(true);
+
+        SCIMUserManager userManager = new SCIMUserManager(mockedUserStoreManager, mockClaimMetadataManagementService,
+                tenantDomain);
+
+        assertEquals(userManager.getEnterpriseUserSchema().size(), 2);
```

Comment:
if possible shall we add another test case to cover the scenario where schema extensions are disabled?

Reply:
It is possible and added a test case